### ISSUE TITLE
Ability to self-host JIRA-GHE plugin ?

### ISSUE THREAD

**Issue by jsire-ledger** [atlassian/github-for-jira#1724]
Hi team !

For a particularly sensible project, we would like to be able to push event from Github Enterprise to JIRA.

We analyzed the already existing Github for Jira plugin, but it appears that it's not compliant with our security policies as it is hosted in Atlassian Cloud, ... and push too many info to Jira (like GH PRs for instance).

--> Is it possible to run the GH-for-Jira app on a server outside of JIRA cloud, ie. to self-hosted the app (note: we have a Jira cloud offer, and don t want to have a self-managed solution).
_We noticed there's a developer mode but it doesn't seem to be designed to be used in prod on our own servers._

Thanks a lot for your help :)

**Comment by mboudreau** [comment]
Just so I'm clear: you have Jira Cloud and yet don't trust Atlassian's Cloud for the integration?

**Comment by jsire-ledger** [comment]
Hi @mboudreau,

That's correct, at least for this very sensitive project only.

What's an issue for us in that case is that webhook gets all the PR and issue data without any way to filter what the app has access to.
We would like to know if there is any way to amend the plugin (and reducing what is pushed to Jira basically) and host the amended plugin within our servers.

**Comment by bgvozdev** [comment]
Hey @jsire-ledger , just a heads up that we are leaning towards reverse proxy solution (https://github.com/atlassian/github-for-jira/issues/1810 ). Potentially, you could've implement a smart reverse proxy that would manipulate the payload and sanitise/exclude sensitive data. Is it something that could've been of your interest?

**Comment by jsire-ledger** [comment]
Thanks a lot for the heads-up.

Last time we talked with @gxueatlassian, we realized that your solution/apps needs to fetch information from our GitHub instance to grab more details (like IDs of different repos, app installation details). So for that to happen, the app needs to "talk" to our GitHub instance, which requires an IP address ( and/or via public DNS ).

However, we aren't able to provide public access to our on-prem GitHub instances. We are trying to figure out another way to work around that. 

We will keep monitoring the progress of the issue #1810 as it might match our requirements eventually.
